cmd/root.go

```diff
@@ -217,199 +87,56 @@ var RootCmd = &cobra.Command{
-	Orderable   bool     `json:"orderable"`
-	Navigable   bool     `json:"navigable"`
-	Searchable  bool     `json:"searchable"`
-	ClauseNames []string `json:"clauseNames"`
-	Schema      struct {
-		Type     string `json:"type"`
-		System   string `json:"system,omitempty"`
-		Items    string `json:"items,omitempty"`
-		Custom   string `json:"custom,omitempty"`
-		CustomID int    `json:"customId,omitempty"`
-	} `json:"schema,omitempty"`
-}
-
-// getFieldIDs requests the metadata of every issue field in the JIRA
-// project, and saves the IDs of the custom fields used by issue-sync.
-func getFieldIDs(client jira.Client) error {
-	log.Debug("Collecting field IDs.")
-	req, err := client.NewRequest("GET", "/rest/api/2/field", nil)
-	if err != nil {
-		return err
-	}
-	fields := new([]JIRAField)
-
-	_, _, err = makeJIRARequest(func() (interface{}, *jira.Response, error) {
-		res, err := client.Do(req, fields)
-		return nil, res, err
-	})
-	if err != nil {
-		return err
-	}
-
-	for _, field := range *fields {
-		switch field.Name {
-		case "GitHub ID":
-			ghIDFieldID = fmt.Sprint(field.Schema.CustomID)
-		case "GitHub Number":
-			ghNumFieldID = fmt.Sprint(field.Schema.CustomID)
-		case "GitHub Labels":
-			ghLabelsFieldID = fmt.Sprint(field.Schema.CustomID)
-		case "GitHub Status":
-			ghStatusFieldID = fmt.Sprint(field.Schema.CustomID)
-		case "GitHub Reporter":
-			ghReporterFieldID = fmt.Sprint(field.Schema.CustomID)
-		case "Last Issue-Sync Update":
-			isLastUpdateFieldID = fmt.Sprint(field.Schema.CustomID)
-		}
-	}
-
-	if ghIDFieldID == "" {
-		return errors.New("Could not find ID of 'GitHub ID' custom field. Check that it is named correctly.")
-	} else if ghNumFieldID == "" {
-		return errors.New("Could not find ID of 'GitHub Number' custom field. Check that it is named correctly.")
-	} else if ghLabelsFieldID == "" {
-		return errors.New("Could not find ID of 'Github Labels' custom field. Check that it is named correctly.")
-	} else if ghStatusFieldID == "" {
-		return errors.New("Could not find ID of 'Github Status' custom field. Check that it is named correctly.")
-	} else if ghReporterFieldID == "" {
-		return errors.New("Could not find ID of 'Github Reporter' custom field. Check that it is named correctly.")
-	} else if isLastUpdateFieldID == "" {
-		return errors.New("Could not find ID of 'Last Issue-Sync Update' custom field. Check that it is named correctly.")
-	}
-
-	log.Debug("All fields have been checked.")
-
-	return nil
-}
-
 // compareIssues gets the list of GitHub issues updated since the `since` date,
 // gets the list of JIRA issues which have GitHub ID custom fields in that list,
 // then matches each one. If a JIRA issue already exists for a given GitHub issue,
 // it calls updateIssue; if no JIRA issue already exists, it calls createIssue.
-func compareIssues(ghClient github.Client, jiraClient jira.Client) error {
+func compareIssues(config lib.Config, ghClient github.Client, jiraClient jira.Client) error {
```

Comment:
all of these functions that are *not* directly concerned with configuring and launching the command should be outside of this file and package. this function should be in lib.